firmware/library/L2_HAL/temperature.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include "L1_Drivers/i2c.hpp"
+
+class TemperatureInterface
+{
+ public:
+  virtual bool Initialize()                       = 0;
+  virtual int GetCelsius()                        = 0;
+  virtual int GetFahrenheit(int temperature_data) = 0;
+};
+
+class Temperature : public TemperatureInterface
+{
+ public:
+  bool Initialize() override
+  {
+  temperature_device_.Initialize();
+  uint8_t device_id = 0x00;
+  // Checks to see if the value read by I2c
+  // for the Temperature Chip ID register
+  // is the same as the Base ID (0x14).
+  temperature_device_.WriteThenRead(kTemperatureAddress, &kTemperatureId,
+  1, &device_id, 1);
+  return(device_id == kMemoryMapAddress);
+  }
+
+  int GetCelsius() override
+  {
+  uint8_t const kOneBurst [2]   = { 0xC4, 0x04 };
+  uint8_t const kAutoincBit [2] = { 0xC5, 0x01 };
+  // The register will enable the device to collect data once
+  // and automatically sets the stop bit to 0 (2nd bit).
+  status_ = temperature_device_.Write(kTemperatureAddress, kOneBurst, 2);
+  // Auto increments I2c register address pointer.
+  status_ = temperature_device_.Write(kTemperatureAddress, kAutoincBit, 2);
+  uint8_t least_significant_register = 0xC2;
+  uint8_t most_significant_register  = 0xC1;
+  temperature_device_.WriteThenRead(kTemperatureAddress,
+ &most_significant_register, 1, &most_significant_register, 1);
+  temperature_device_.Read(kTemperatureAddress, &least_significant_register, 1);
+
+  int data = 0;
+  data = ((most_significant_register & 0x7F) << 8) | least_significant_register;
+  // The required computation after bit shifting.
+  return (((data - 16384) / 160 ) + 55);
+  }
+
+  int GetFahrenheit(int temperature_data)
+  {
+  // Formula used to convert Celsius to Fahrenheit
+  return (((temperature_data - 32) * 5) / 9 );
+  }
+
+ private:
+  I2c temperature_device_;
+  Status status_;
+  static constexpr uint8_t kTemperatureAddress = 0x31;
+  static constexpr uint8_t kTemperatureId      = 0xC0;
+  static constexpr uint8_t kMemoryMapAddress   = 0x14;
```

Comment:
static and constexpr should be at the top.

https://google.github.io/styleguide/cppguide.html#Class_Format